zap/src/main/java/org/zaproxy/zap/extension/history/HistoryFilterPlusDialog.java

```diff
@@ -119,11 +124,14 @@ private void initialize() {
         this.setContentPane(getJPanel());
         this.setVisible(false);
         this.setTitle(Constant.messages.getString("history.filter.title"));
+
         if (Model.getSingleton().getOptionsParam().getViewParam().getWmUiHandlingOption() == 0) {
             this.setSize(600, 300);
         }
+
         centreDialog();
         this.getRootPane().setDefaultButton(btnApply);
+        this.setMinimumSize(new Dimension(1000, 600));
```

Comment:
The minimum size should not be set as a rule (e.g. VM with 800x600 resolution would have problems). Maybe you meant the size and even then, better if packed and let the layout manager size the components to their preferred size.

Reply:
Default packing looks terrible, you need to expand the window every single time you run ZAP to see all values. It is not convenient. Also, using tools like ZAP with resolution 800x600 is not something you would find convenient. I can redesign window but I believe we need field values visible without need of expanding the window.

Reply:
If the packing looks terrible then preferred sizes could/should be specified, or other mechanisms leveraged to make it work nicely (note: I am not and do not claim to be a java UI expert, far from it). Setting the minimum completely takes control away from the user, and potentially makes things unusable if they happen to be on underspeced systems/VMs.